File: app/app/src/main/java/me/williamhester/reddit/ui/viewholders/SubmissionLinkViewHolder.java
package me.williamhester.reddit.ui.viewholders;

import android.view.View;
import android.widget.TextView;

import butterknife.ButterKnife;
import me.williamhester.reddit.R;
import me.williamhester.reddit.models.Submission;
import me.williamhester.reddit.ui.VotableCallbacks;

/**
 * Created by william on 6/13/16.
 */
public class SubmissionLinkViewHolder extends SubmissionViewHolder {
  private final TextView linkText;

  public SubmissionLinkViewHolder(View itemView, final VotableCallbacks clickListener) {
    super(itemView, clickListener);

    linkText = ButterKnife.findById(itemView, R.id.link_text);
    linkText.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        // TODO: click submission link
//        clickListener.onLinkClicked();
      }
    });
  }

  @Override
  public void setContent(Submission item) {
    super.setContent(item);

    linkText.setText(item.getDomain());
  }
}

File: app/app/src/main/java/me/williamhester/reddit/ui/viewholders/SubmissionLinkViewHolder.kt
package me.williamhester.reddit.ui.viewholders

import android.view.View
import android.widget.TextView

import me.williamhester.reddit.R
import me.williamhester.reddit.models.Submission
import me.williamhester.reddit.ui.VotableCallbacks

/** A ViewHolder for a Submission object. */
class SubmissionLinkViewHolder(
    itemView: View,
    clickListener: VotableCallbacks
) : SubmissionViewHolder(itemView, clickListener) {

  private val linkText: TextView = itemView.findViewById(R.id.link_text)

  init {
    linkText.setOnClickListener {
      // TODO: click submission link
      //        clickListener.onLinkClicked();
    }
  }

  override fun setContent(item: Submission) {
    super.setContent(item)

    linkText.text = item.domain
  }
}
